### DISCUSSION TITLE
Backup/Restore in special use case

### DISCUSSION THREAD

**Discussion by flachica** [Tecnativa/doodba#483] [Doodba - The template] [answered]
I have a special use case and I would like to know if it is possible to restore a copy in my Doodba development environment

The production system does not use Doodba. We haven't gotten around to setting it up yet (but I'll get there). I was wondering if there is a possibility of:

* Create system copy from "<odoo>/web/database/manager"
* Take the generated file
* Locate it in some local directory
* Proceed with an invoke restore-snapshot or smiliar

According to and interpreted when reading the [documentation](https://github.com/Tecnativa/doodba-copier-template/blob/main/docs/daily-usage.md#backups), I think that this is not possible (Backups are only available in the production environment) from the base but seeing a little the code I think that there is some possibility

Has this situation occurred to you? Any recommendations?

Today I can do it through "<odoo>/web/database/manager" but as the database grows this will not work

**Comment by rquevedo** [comment, marked as answer]
@flachica First you need to backup your production database from terminal, directly from postgresql. Use database/manager is not advised if you are in production environment and as you say if bbdd gets big is useless. Ok, when you have a backup.dump you can run this inside your doodba folder:
`docker exec -i 100_db_1 pg_restore -U odoo -d devel -v < /path/to/backup/backup.dump`

Where '100_db_1' is the name of your db container and 'devel' is the name of database(if you follow directives and your environment is develop)

**Reply by antoniospneto** [reply]
@flachica @rquevedo 

Good to remember that you have to copy the files from the **filestore** folder too.

**Reply by pleirb** [reply]
> @flachica @rquevedo
> 
> Es bueno recordar que también debe copiar los archivos de la carpeta del almacén **de archivos** .

And what would be the steps to carry out this neatly? Use the volumes in the docker-compose.yml file? Does it have to be yes or yes in development mode?

**Reply by rquevedo** [reply]
> > @flachica @rquevedo
> > Es bueno recordar que también debe copiar los archivos de la carpeta del almacén **de archivos** .
> 
> And what would be the steps to carry out this neatly? Use the volumes in the docker-compose.yml file? Does it have to be yes or yes in development mode?

No, this can be done in production environment too, but is less frequent as in production you have a backup container and restore is easier. If you plan to do it in production, just remember to change database name.

**Reply by pleirb** [reply]
It has been more practical for me to edit the common.yaml and change the line filestore:/var/lib/odoo:z to ./odoo/odoo:/var/lib/odoo:rw,z
In this way I have access to the filestore from the project folder and I assume that it would be valid for any environment
